$NetBSD: patch-glib_meson.build,v 1.5 2019/11/14 11:40:30 wiz Exp $

- On Darwin, allow building without Cocoa.
- Don't bother with gtester as it's slated for deprecation anyway:
  https://gitlab.gnome.org/GNOME/glib/issues/1441

--- glib/meson.build.orig	2019-10-04 11:44:24.000000000 +0000
+++ glib/meson.build
@@ -316,7 +316,7 @@ else
   platform_deps = []
 endif
 
-if host_system == 'darwin'
+if host_system == 'darwin' and glib_have_cocoa
   glib_sources += files('gosxutils.m')
 endif
 
@@ -410,25 +410,8 @@ if host_system == 'windows'
       include_directories : configinc,
       dependencies : [libglib_dep])
   endif
-else
-  gtester = executable('gtester', 'gtester.c',
-    install : true,
-    c_args : ['-UG_DISABLE_ASSERT'],
-    include_directories : configinc,
-    dependencies : [libglib_dep])
 endif
 
-report_conf = configuration_data()
-report_conf.set('GLIB_VERSION', glib_version)
-report_conf.set('PYTHON', python_name)
-configure_file(
-  input: 'gtester-report.in',
-  output: 'gtester-report',
-  install_dir: get_option('bindir'),
-  configuration: report_conf,
-  install_mode: 'rwxr-xr-x'
-)
-
 install_data('glib_gdb.py', install_dir : join_paths(glib_pkgdatadir, 'gdb'))
 
 gdb_conf = configuration_data()
